### PR TITLE
Infra cleanup part1

### PR DESCRIPTION
**Pull request by tbody-cfs**
* Moves default units to a separate YAML file
* Removes `named_options.Algorithms`
* Makes an autogenerated `algorithms.yaml` file describing the inputs and outputs of each algorithm
* Make a wrapper to make an Algorithm for a function

### CODE REVIEW

**Comment by tbody-cfs** [comment]
Replaces #39

**Comment by nelsonand** [comment]
We will check this!

**Comment by nelsonand** [comment]
@aarooshgr This is the new algorithm format we talked about

File: cfspopcon/unit_handling/default_units.py [hunk 1]
```diff
@@ -1,145 +1,42 @@
...
+
+
+def read_default_units() -> dict[str, str]:
+    """Read in the default_units.yaml file."""
+    with as_file(files("cfspopcon").joinpath("default_units.yaml")) as filepath:
+        with open(filepath) as f:
+            units_dictionary: dict[str, str] = yaml.safe_load(f)
+
+    check_units_are_valid(units_dictionary)
+    return units_dictionary
```
Comment (hassec) [inline cfspopcon/unit_handling/default_units.py:36]: 👍

File: tests/test_docs.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import pytest
```
Comment (hassec) [inline tests/test_docs.py:1]: Oh neat, great idea! 👍

File: cfspopcon/algorithm_class.py [hunk 3]
```diff
@@ -128,19 +137,63 @@ def wrapped_function(**kwargs: Any) -> dict:
...
         return cls(wrapped_function, return_keys, name=name)
 
+    @classmethod
+    def register_algorithm(
+        cls, return_keys: list[str], name: Optional[str] = None, skip_unit_conversion: bool = False
+    ) -> GenericFunctionType:
+        """Decorate a function and turn it into an Algorithm. Usage: @Algorithm.register_algorithm(return_keys=["..."])."""  # noqa: D402
+
+        def function_wrapper(func: GenericFunctionType) -> GenericFunctionType:
+            Algorithm.from_single_function(func, return_keys=return_keys, name=name, skip_unit_conversion=skip_unit_conversion)
```
Comment (hassec) [inline cfspopcon/algorithm_class.py:156]: Should we check if `name is None` and then use `func.__name__` instead?

File: cfspopcon/__init__.py [hunk 4]
```diff
@@ -23,11 +23,12 @@
     wraps_ufunc,
 )
 
+Algorithm.write_yaml()
```
[the remark this answers is not in the document]
Comment (tbody-cfs) [reply cfspopcon/__init__.py:26]: Resolve in #41

File: cfspopcon/algorithm_class.py (hunk 3 above)
Comment (tbody-cfs) [reply cfspopcon/algorithm_class.py:156]: Resolve in #41

**Review by hassec — APPROVED** [review]

**Pull request merged**